docs/src/tutorials/quadcopter.md

```diff
@@ -0,0 +1,256 @@
+# Optimal Control of a Quadcopter
+
+By: Rishi Mandyam
+
+This tutorial notebook is an introduction to the graph-based modeling framework 
+Plasmo.jl (Platform for Scalable Modeling and Optimization) for JuMP 
+(Julia for Mathematical Programming).
+
+The following problem comes from the paper of Na, Shin,  Anitescu, and Zavala (available [here](https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=9840913)).
+
+A quadcopter operates in 3-D space with positions $(x, y, z)$ and angles ($\gamma$, $\beta$, and $\alpha$). 
+$g$ is the graviational constant. The set of state variables at time $t$ are treated as $\boldsymbol{x}_t = (x, y, z, \dot{x}, \dot{y}, \dot{z}, \gamma, \beta, \alpha)$. 
+The input variables at time $t$ are $\boldsymbol{u}_t = (a, \omega_x, \omega_y, \omega_z)$. 
+
+The quadcopter control problem can be written as an optimization problem as:
+
+```math
+\begin{align*}
+    \min &\; \phi(t) := \int_{0}^T \frac{1}{2} (\boldsymbol{x}(t) - \boldsymbol{x}(t)^{ref})^\top Q (\boldsymbol{x}(t) - \boldsymbol{x}(t)^{ref}) + \boldsymbol{u}(t)^\top R \boldsymbol{u}(t) dt \\
+    \textrm{s.t.} &\; \frac{d^2x}{dt^2} = a (\cos(\gamma) \sin( \beta) \cos (\alpha)  + \sin (\gamma) \sin (\alpha)) \\
+&\; \frac{d^2 y}{dt^2} = a (\cos (\gamma) \sin (\beta) \sin (\alpha) - \sin (\gamma) \cos (\alpha)) \\
+&\; \frac{d^2 z}{dt^2} = a \cos (\gamma) \cos (\beta) - g \\
+&\; \frac{d\gamma}{dt} = (\omega_x \cos (\gamma) + \omega_y \sin (\gamma)) / \cos (\beta)\\
+&\;\frac{d\beta}{dt} = -\omega_x \sin (\gamma) + \omega_y \cos (\gamma) \\
+&\;\frac{d\alpha}{dt} = \omega_x \cos (\gamma) \tan (\beta) + \omega_y \sin (\gamma) \tan (\beta) + \omega_z
+\end{align*}
+```
+
+We will model this problem in Plasmo by discretizing the problem into finite time points and representing each time point with a node. 
+
+### 1. Import Packages
+To begin, we will import and use the necessary packages
+
+```julia 
+using JuMP
+using Plasmo
+using Ipopt
+using Plots
+using LinearAlgebra
+```
+
+
+### 2. Function Design
```

Comment:
I would suggest splitting the model build into multiple functions as opposed to one splitting one function into multiple snippets. At the end, you could call something like `build_quadcopter_model` that assembles the whole optigraph. To make things easier, you could define each function to accept an optigraph as input and do their modifications. So something like:
```julia
graph = OptiGraph()
data = initialize_data(N)
add_variables(graph, data)
add_initial_condition(graph, data)
add_link_constraints(graph, data)
```